infra.bs

```diff
@@ -535,14 +538,17 @@ contains, in the range 0x61 (a) to 0x7A (z), inclusive, by 0x20.
 
 <p>To <dfn export>isomorphic decode</dfn> a <a>byte sequence</a> <var>input</var>, return a
 <a>string</a> whose <a for=string>code point length</a> is equal to <var>input</var>'s
-<a for="byte sequence">length</a> and whose <a>code points</a> have the same values as
-<var>input</var>'s <a>bytes</a>, in the same order.
+<a for="byte sequence">length</a> and whose <a>code points</a> have the same
+<a for="code point">values</a> as the <a for=byte>values</a> of <var>input</var>'s <a>bytes</a>, in
+the same order.
 
 
 <h3 id=code-points>Code points</h3>
 
 <p>A <dfn export lt="code point|character">code point</dfn> is a Unicode code point and is
-represented as a four-to-six digit hexadecimal number, typically prefixed with "U+".
+represented as "U+" followed by four-to-six <a>ASCII upper hex digits</a>, in the range U+0000 to
```

Comment:
Is this too circular? Not really sure how else to do this.

Reply:
Maybe do it the other way around? Perhaps:

> A `code point` is a Unicode code point, whose `value` is an integer between 0 and 0x10FFFF inclusive and represented as the string `U+` followed by four to six `ASCII upper hex digits`.

Reply:
I think that still has the same issue, as ASCII upper hex digits are defined using the U+ convention.

Reply:
Avoiding the circularity seems like it would lead to double-defining ASCII upper hex digits and reduce the overall clarity. I don't think people have a hard time with the U+ syntax so much as they have a difficult time with the difference between USVs/code points vs. code units (particularly the UTF-16 variety).

Reply:
I don't think we can really avoid circularity in Infra without making it significantly more opaque. See previous discussions at https://github.com/whatwg/infra/issues/230#issuecomment-445774344